### DISCUSSION TITLE
Limit the total number of search results

### DISCUSSION THREAD

**Discussion by SinghGurd** [microsoft-search/pnp-modern-search#3165] [General]
Is there a way to limit the count of search results being returned e.g. only fetch top 5000 results? We are already using paging and only showing 10 results per page. However, user has the ability to navigate pages and when they click on last page (which has a page number of 359000), they are getting an error like this below screenshot. I think this may be due to large page number being sent in search query.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/26758133/2cf6610d-7457-40ba-9077-1efc41a585fe)

**Comment by wobba** [comment]
I would not expect any end user to click the last page of such a large result set.

You can modify the template and pass in some max number to the `pnp-pagination` control. Something like this:

```
{{#gte @root.data.totalItemsCount 5000}}
            <pnp-pagination 
                    data-total-items="5000" 
    {/gte}
```

**Reply by SinghGurd** [reply]
That's great...thank you @wobba

**Reply by wobba** [reply]
Let me know if this works as I didn't get to test it myself.

**Reply by SinghGurd** [reply]
It makes sense to think that no one wants to go to last page of such a large data set. Hence, we decided to take the easy route and simply use the built-in web part properties to hide the first and last buttons. I think it would still be beneficial to have some sort of setting - query or web part property - to limit to maximum number of results as that could also make search perform a bit better.
